src/Driver.cpp

```diff
@@ -1,598 +1,558 @@
 #include "Driver.hpp"
 #include <iostream>
 #include <sstream>
-#include "Exceptions.hpp"
+//#include "Exceptions.hpp"
+
 using namespace usbl_evologics;
+
 Driver::Driver()
-    : iodrivers_base::Driver(1000)
-{
-}
-
-ConnectionStatus Driver::getConnectionStatus(){
-    sendWithLineEnding("+++AT?S");
-    return (ConnectionStatus) (UsblParser::parseConnectionStatus(waitSynchronousString("AT?S"))); 
-}
-
-DeviceSettings Driver::getDeviceSettings(){
-    DeviceSettings ds;
-    ds.sourceLevel = getSourceLevel();
-    ds.sourceLevelControl = getSourceLevelControl();
-    ds.lowGain = getLowGain();
-    ds.carrierWaveformId = getCarrierWaveformId();
-    ds.localAddress = getLocalAddress();
-    ds.clusterSize = getClusterSize();
-    ds.packetTime = getPacketTime();
-    ds.retryCount = getRetryCount();
-    ds.retryTimeout = getRetryTimeout();
-    ds.speedSound = getSpeedSound(); 
-    ds.imRetry = getImRetry();
-    ds.idleTimeout = getIdleTimeout();
-    return ds;
-}
-
-DeviceStats Driver::getDeviceStatus(){
-    DeviceStats  device_stats;
-    device_stats.dropCount = getDropCounter();
-    device_stats.overflowCount = getOverflowCounter();
-    device_stats.localRemoteBitrate = getLocalRemoteBitrate();
-    device_stats.remoteLocalBitrate = getRemoteLocalBitrate();
-    device_stats.receivedSignalStrengthIndicator = getReceivedSignalStrengthIndicator();
-    device_stats.signalIntegrityLevel = getSignalIntegrityLevel();
-    device_stats.propagationTime = getPropagationTime();
-    device_stats.relativeVelocity = getRelativeVelocity();
-    return device_stats;
-}
-
-int Driver::getIntValue(std::string value_name){
-    std::stringstream ss;
-    ss << "+++" << value_name;
-    std::string s = ss.str();
-    sendWithLineEnding(s);
-    return waitSynchronousInt(s.substr(3, s.size()));
+: iodrivers_base::Driver(max_packet_size)
+{
+    mode = DATA;
 }
-DeliveryStatus Driver::getInstantMessageDeliveryStatus(){
-    return currentInstantMessage.deliveryStatus;
+
+Driver::~Driver()
+{
 }
 
-Position Driver::getPosition(bool x){
-    //TODO maybe make both variants possible polling and asynch messages 
-    //important a
+//bool Driver::sendData(bool &mailCommand)
+//{
+//    bool ret = sendCommand(mailCommand);
+//    if(mode == DATA)
+//    {
+//        ret |= sendRawData();
+//    }
+//    return ret;
+//}
 
-    if(reverse_mode == REVERSE_POSITION_RECEIVER){
-	std::cout << "usbl driver.cpp is position receiver!\n";
-    std::cout << "now returning X: " << current_position.x << std::endl;
-    std::cout << "now returning Y: " << current_position.y << std::endl;
-    std::cout << "now returning Z: " << current_position.z << std::endl;
+WaitResponse Driver::sendCommand(void)
+{
+    if(!queueCommand.empty())
+    {
+        std::string command = queueCommand.front();
 
-        new_position_available = false;
+        std::cout << "Write Line: " << command << std::endl;
```

Comment:
Don't check in code that has debug output like this. If you really want to have debug output, use Rock's logger (http://rock-robotics.org/master/documentation/debugging/index.html)